nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/src/main/java/org/apache/nifi/prometheus/util/PrometheusMetricsUtil.java

```diff
@@ -101,20 +98,6 @@ public static CollectorRegistry createNifiMetrics(NiFiMetricsRegistry nifiMetric
         final String componentId = StringUtils.isEmpty(status.getId()) ? DEFAULT_LABEL_STRING : status.getId();
         final String componentName = StringUtils.isEmpty(status.getName()) ? DEFAULT_LABEL_STRING : status.getName();
 
-        // Clear all collectors to deal with removed/renamed components -- for root PG only
-        if("RootProcessGroup".equals(componentType)) {
```

Comment:
Thanks for the response.
Removing this code does not cause a regression in PrometheusReportingTask. The correct behavior is confirmed by the existing tests starting on line 142 in PrometheusReportingTaskIT.java as well as manually confirmed.
The removed code is no longer needed because a new NiFiMetricsRegistry is created (line 142 in PrometheusReportingTask.java) before each call to createNifiMetrics.
This change was introduced in NIFI-7379 Support multiple instances of Prometheus registries/metrics.

In hindsight, I should have had the removed code as a separate ticket as it isn't required to be removed to fix the bug I reported in NIFI-8272. Although it is all related so I did the work together.